tests/baselines/reference/controlFlowInstanceof.types

```diff
@@ -20,16 +20,16 @@ function f1(s: Set<string> | Set<number>) {
 >Set : SetConstructor
 
         s;  // Set<number>
->s : Set<number>
+>s : Set<number> & Set<any>
     }
     s;  // Set<number>
->s : Set<number>
+>s : Set<number> & Set<any>

```

Comment:
Are these cases where TS would widen to be an `any` instead of the original `number`, this feels like something we don't want

Reply:
Yes, when accessing the parametrising type you'll end up getting `number & any` which is going to swallow `number` - this is definitely a bad change. The problem is that before it was relying on the assignability check to strip out the `any` case, but now we don't have that.

I think I remember that there was a similar issue that came up in conditional type simplification, where `T extends Foo<any> ? A : B` was simplified to `T & Foo<any>`, causing similar problems. @weswigham what was the solution here?